pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/MinionInstancesCleanupTask.java

```diff
@@ -54,8 +54,11 @@ public MinionInstancesCleanupTask(PinotHelixResourceManager pinotHelixResourceMa
         controllerConf.getMinionInstancesCleanupTaskMinOfflineTimeBeforeDeletionInSeconds() * 1000L;
   }
 
+  /**
+   * @param filter Currently not used, but can be used to specify how this task should be run.
+   */
   @Override
-  protected void runTask() {
+  protected void runTask(String filter) {
```

Comment:
Instead of adding a filter that is to be interpreted by each subclass, please add  `shouldRunForTable()` in base periodic task, and use that in all the sub-classes (that operate on a per-table basis). You can then add all logic there.

Reply:
I thought about that, but the issue there is that if we put `shouldRunForTable()` in base class, it may not apply to all subclasses? How about adding a function `setAttribute(String key, String value)` in the base class and this could be called as `setAttribute("tablename", tableName)`? For now, the only attribute, we would allow is "tablename". This way there is no strong association between table name and all the task classes which may not use the table name?

Reply:
It will apply to `ControllerPeriodicTask`. The other tasks are not table-based, and so it is ok

Reply:
Looking into this...